io/include/detray/io/common/geometry_writer.hpp

```diff
@@ -245,6 +217,66 @@ class geometry_writer : public writer_interface<detector_t> {
                 typename material_group_t::value_type>(index);
         }
     };
+
+    /// Retrieve @c acc_links_payload from surface_tore collection
+    struct get_acc_link_payload {
+        template <typename acc_group_t, typename index_t>
+        constexpr inline auto operator()(const acc_group_t&,
+                                         const index_t& index) const {
+
+            using accelerator_t = typename acc_group_t::value_type;
+
+            if constexpr (detail::is_grid_v<accelerator_t>) {
+                constexpr auto id{
+                    detail::get_grid_id(accelerator_t::local_frame())};
+                return geometry_writer<detector_t>::serialize(id, index);
+            } else {
+                // This functor is only called for accelerator data structures
+                // that are not 'brute force'
+                return geometry_writer<detector_t>::serialize(
+                    acc_links_payload::acc_type::unknown, index);
+            }
+        }
+    };
+
+    /// Infer the IO shape id from the shape type
+    template <typename shape_t>
+    static constexpr io::detail::mask_shape get_shape_id() {
+
+        // Find the correct shape IO id;
```

Comment:
Could one solve this by rolling through a tuple?

Reply:
I don't know how it slipped my mind, but I had actually written a tool for this ages ago